### PR TITLE
Port of uri_for from Psr7\uri_for to Utils::uri_for

### PR DESCRIPTION
**Pull request by hkmekala**
As part of the latest release in Guzzle\Psr7, they have removed the Functions API in order to fix the conflicts between global and local copies in their library this Pull Request uses the latest Utils class which is about to be introduced in Guzzle\Psr7 2.0.0

*Issue #, if available:*
https://github.com/aws/aws-sdk-php/issues/2213

*Description of changes:*

The function uses the new Utils class to call the static methods to support the latest release of Guzzle\Psr 7 2.0.0

https://github.com/guzzle/psr7/blob/ff1b0eae41c6bc0b03ef657f29d8a9eea7ee75eb/README.md#upgrading-from-function-api

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by VincentLanglet — APPROVED** [review]

**Comment by SamRemis** [comment]
Hi @hkmekala,
Thank you so much for your PR, it looks like all of the automated tests are going to pass which is a good sign :) I'll take a closer look and give feedback or merge ASAP.  In the meantime, could you add a changelog file to document the changes?

**Comment by hkmekala** [comment]
Hi @SamRemis,
I have added the Changelog.

**Review by SamRemis — COMMENTED** [review]
Thank you for your contribution and being so thorough.  It looks good and has passed the github actions, so I'm confident that this will be a safe change to merge.  Just for my sanity's sake though, I can't find any documentation online about the switch to the utils:: methods, other than just stating that they're now static and the old ones are deprecated.  

Is there any guidance from PSR7 that says that the behavior of the functions will stay the same when switching to the static methods

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## 3.175.4 - 2021-03-25
```
[the remark this answers is not in the document]
Comment (hkmekala) [reply CHANGELOG.md]: Sorry @SamRemis, I have added the file in the ./.changes. Please review my PR. Thank you.

**Comment by hkmekala** [comment]
@SamRemis It's regarding the failed Testsuite. The JSON validation has failed. it might be coz I have placed the category as empty string since I have updated a lot of services.

> If your update is for core components of the SDK, the category field should exist with the value set to an empty string ""

File: .changes/3.178.6 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[
```
Comment (SamRemis) [inline .changes/3.178.6:1]: this file should be at a different location: 
.changes/nextrelease/bugfix-migration-to-static-psr7-functions.json.

Personally I would think this is an enhancement and change the type to enchancement and name it 
.changes/nextrelease/enhancement-migration-to-static-psr7-functions.json.

Comment (hkmekala) [reply .changes/3.178.6]: @SamRemis, I have added the file .changes/nextrelease/enhancement-migration-to-static-psr7-functions.json and changed the type to enhancement.

**Pull request merged**